packages/network-console-shared/src/file_io/interfaces.ts

```diff
@@ -112,3 +116,146 @@ export interface ICollectionItemAdapter extends ICollectionEntryAdapter {
 
     readonly request: INetConsoleRequest;
 }
+
+/**
+ * Represents a file format for Environments, which are collections of variables
+ * which can be substituted into requests.
+ */
+export interface IEnvironmentFormat {
+    /**
+     * Gets an internal ID for the format.
+     */
+    readonly formatId: string;
+    /**
+     * Gets whether this format, or any of the objects it produces, are writable.
+     * If not, attempting to write to any of them will raise an exception.
+     */
+    readonly canWrite: boolean;
+
+    /**
+     * Creates a new, empty environment container in this format. If `canWrite` is false,
+     * this method may throw an error.

```

Comment:
may or will?

Reply:
The contract is "may", and is left up to the implementation. In practice, presently, it's "will".